app/controllers/agents_controller.rb

```diff
@@ -220,9 +220,9 @@ def redirect_back(message, options = {})
   end
 
   def build_agent
-    @agent = Agent.build_for_type(params[:agent].delete(:type),
+    @agent = Agent.build_for_type(agent_params.delete(:type),
```

Comment:
I have written `agent_params` so that it is a drop in replacement for `params[:agent]` (which I assume to be true for all `*_params` helper, but maybe that's just my "convention"). Do you find the memorization confusing or the way `agent_params` is used?